src/analyses/interval_domain.h

```diff
@@ -40,10 +40,15 @@ class interval_domaint:public ai_domain_baset
     const ai_baset &ai,
     const namespacet &ns) const final;
 
+  bool join(const interval_domaint &b);
```

Comment:
Ok, seems fair. What irritated me a bit about this was the two verbs with very similar meaning (in this context). Using different verbs suggests that they do different things, but then the difference is very hard to find out (especially when `merge` just calls `join`). Can we maybe only use one of the two, or else have a comment that explains the difference?